internal/acctest/template/generator_framework_test.go

```diff
@@ -0,0 +1,266 @@
+package template
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework-timeouts/resource/timeouts"
+	datasourceschema "github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	resourceschema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFrameworkGenerateTemplate(t *testing.T) {
+	ctx := context.Background()
+
+	tests := []struct {
+		name         string
+		schema       interface{}
+		resourceType string
+		kind         ResourceKind
+		want         string
+	}{
+		{
+			name: "resource with boolean fields",
+			schema: resourceschema.Schema{
+				Attributes: map[string]resourceschema.Attribute{
+					"enable_feature": resourceschema.BoolAttribute{
+						Required: true,
+					},
+					"disable_feature": resourceschema.BoolAttribute{
+						Optional: true,
+					},
+					"settings": resourceschema.SingleNestedAttribute{
+						Required: true,
+						Attributes: map[string]resourceschema.Attribute{
+							"feature_one": resourceschema.BoolAttribute{
+								Required: true,
+							},
+							"feature_two": resourceschema.BoolAttribute{
+								Optional: true,
+							},
+						},
+					},
+				},
+			},
+			resourceType: "test_resource",
+			kind:         ResourceKindResource,
+			want: `resource "test_resource" "{{ required .resource_name }}" {
```

Comment:
nit: Even if it is not totally isolated, maybe testing the final result (TF code) would be beneficial and easier for maintaining and understanding. SO maybe it makes sense to add necessary functions before rendering.